### DISCUSSION TITLE
Flashing the ESP32-S3 DevKitC RGB LED

### DISCUSSION THREAD

**Discussion by smhodge42** [micropython/micropython#13177] [Show and tell]
I presume most of us are like me that when faced with a new microcontroller, we like to use the LED equivalent of "hello world". The pyboard has built-in LED's and I always design one into any custom PCB design, programmed in either uPython or C++.  Well, the built-in RBG LED on the new-to-me ESP32-S3-DevKitC-1 board stumped me, having never dealt with an addressable LED before. Plug a fresh-from-the-factory board into power and the RGB LED flashes, so I knew it must be possible. But that code vanished the instant I upgraded the firmware.

It took me a couple of days to figure out the uPython code to do it, largely because I could not find anything on the Internet -- or for that matter in Micropython Documents -- on exactly how to do this supposedly simple task (as a newbie to this board). It retrospect it is pretty simple after you realize/learn that the key is  the "NeoPixel" class.  I'm posting the code here so that it might save others the same trouble. The code endlessly flashes Red-Green-Blue, each color every second. Presumably the code will work on any ESP32-xx DevKit board with an addressable LED and a uPython port, such as the S2, C3 or C6 DevKits.

```
from machine import Pin
from time import sleep
from neopixel import NeoPixel
pin = Pin(48, Pin.OUT)                          # Pin number for v1 of the above DevKitC, use pin 38 for v1.1
np = NeoPixel(pin, 1)                            # "1" = one RGB led on the "led bus"
while (True):
    np[0] = (255,0,0)
    np.write()
    sleep(1)
    np[0] = (0,255,0)
    np.write()
    sleep(1)
    np[0] = (0,0,255)
    np.write()
    sleep(1)
```

**Comment by wilsonfleite** [comment]
Thanks for share!

**Comment by WolleKette** [comment]
The board is available in a version with Micro-USB and one with USB-C ports. I'm pretty sure I was able to get the LED to flash with a board (Micro USB version) some time ago.
But now I'm using a USB-C board and the example doesn't work here. Neither on GPIO 38 nor on pin 48.

Does anyone have an idea what the problem could be?

**Reply by WolleKette** [reply]
I found the solution in a arduino forum [here](https://forum.arduino.cc/t/esp32-s3-onboard-rgb-led/1198754/11). You have to close the RGB-named solder bridge next to the LED on the board.

**Comment by meti-0763** [comment]
thanks I looked for it a lot ❤️❤️🗿

**Comment by bidrohini** [comment]
Thanks for sharing.

**Comment by nex0ma** [comment]
esp32-c3-mini with dual USB:

    rgb = Pin(8,1)
    np = NeoPixel(rgb,1)

**Comment by khalidnk83** [comment]
please use the below code, as this is adafruit neopixle led;


#include <Adafruit_NeoPixel.h> 

#define PIN 48
#define NUMPIXELS 1

Adafruit_NeoPixel pixels(NUMPIXELS, PIN, NEO_GRB + NEO_KHZ800);
#define DELAYVAL 500

void setup () {  
  
  pixels.begin();

}

void loop () {

  pixels.clear();

  for(int i=0; i<NUMPIXELS; i++) {
    pixels.setPixelColor(i, pixels.Color(0, 150, 0));
    pixels.show();
    delay(DELAYVAL);
  }

}

**Reply by scruss** [reply]
that's Wiring/Arduino ... we don't do that here

**Comment by neopostmodern** [comment]
Auto-discovery of PIN number:
```
for pin_number in range(1, 40):
  print(pin_number)
  pin = Pin(pin_number, Pin.OUT)
  np = NeoPixel(pin, 1)  
  np[0] = (255, 255, 255)
  np.write()
  sleep(1)  
```
(stand by and hit `CTRL+C` when the light turns on)

**Comment by easytarget** [comment]
See: https://github.com/orgs/micropython/discussions/16472#discussion-7729579

Example there is for a RP2040 but I have tested the code on a ESP32 C3 devboard too.

On the ESP32-**C3** boards the NeoPixel pin is`GPIO8`; this is also on the silkscreen of the (clone) board I have.

**Comment by kovalick** [comment]
#Be careful since v1.1 uses pin 38, v1 uses pin 48. below for V1.1 

from machine import Pin
from time import sleep
from neopixel import NeoPixel

PIXEL_PIN = 38  # v1.1 uses 38, v1 uses 48

pin = Pin(PIXEL_PIN, Pin.OUT)
np = NeoPixel(pin, 1)

print(f"Starting RGB blink on Pin {PIXEL_PIN}")

while True:
    print("Red")
    np[0] = (255, 0, 0)
    np.write()
    sleep(.5)
    
    print("Green")
    np[0] = (0, 255, 0)
    np.write()
    sleep(.5)
    
    print("Blue")
    np[0] = (0, 0, 255)
    np.write()
    sleep(.5)
